### PR TITLE
Refs #18812 - Update bindings on event queue

### PR DESCRIPTION
**Pull request by chris1984**

### CODE REVIEW

File: manifests/qpid.pp [hunk 1]
```diff
@@ -19,7 +19,7 @@
     logoutput => true,
   } ~>
   exec { 'bind katello entitlements queue to qpid exchange messages that deal with entitlements':
-    command   => "qpid-config --ssl-certificate ${client_cert} --ssl-key ${client_key} -b 'amqps://localhost:5671' bind event ${candlepin_event_queue} '*.*'",
+    command   => "qpid-config --ssl-certificate ${client_cert} --ssl-key ${client_key} -b 'amqps://localhost:5671' bind event ${candlepin_event_queue} entitlement.created && qpid-config --ssl-certificate ${client_cert} --ssl-key ${client_key} -b 'amqps://localhost:5671' bind event ${candlepin_event_queue} entitlement.deleted && qpid-config --ssl-certificate ${client_cert} --ssl-key ${client_key} -b 'amqps://localhost:5671' bind event ${candlepin_event_queue} pool.created && qpid-config --ssl-certificate ${client_cert} --ssl-key ${client_key} -b 'amqps://localhost:5671' bind event ${candlepin_event_queue} pool.deleted qpid-config --ssl-certificate ${client_cert} --ssl-key ${client_key} -b 'amqps://localhost:5671' bind event ${candlepin_event_queue} compliance.created",
```
Comment (stbenjam) [inline manifests/qpid.pp]: I'd make these separate execs so it's clear what one fails if one fails

Comment (chris1984) [inline manifests/qpid.pp]: +1 easier to read too :) didnt know if you wanted them all in one or separate, will update the pr.

**Comment by chris1984** [comment]
@stbenjam thanks for the review, updated the pr based on your suggestions.

**Comment by ehelms** [comment]
These look like they could benefit from a `define` that just passes in the event name to DRY them up a bit and make the list of events more obvious. 

Would be nice if the commit message explained a bit why these are being updated (i.e. split out into specific events).

File: manifests/qpid.pp [hunk 2]
```diff
@@ -18,12 +18,39 @@
     require   => Service['qpidd'],
     logoutput => true,
   } ~>
-  exec { 'bind katello entitlements queue to qpid exchange messages that deal with entitlements':
-    command   => "qpid-config --ssl-certificate ${client_cert} --ssl-key ${client_key} -b 'amqps://localhost:5671' bind event ${candlepin_event_queue} '*.*'",
+  exec { 'bind katello entitlements queue to qpid exchange messages that deal with entitlement creation':
+    command   => "qpid-config --ssl-certificate ${client_cert} --ssl-key ${client_key} -b 'amqps://localhost:5671' bind event ${candlepin_event_queue} entitlement.created",
+    onlyif    => "qpid-config --ssl-certificate ${client_cert} --ssl-key ${client_key} -b 'amqps://localhost:5671' queues ${candlepin_event_queue}",
```
Comment (stbenjam) [inline manifests/qpid.pp]: This is a check to see whether the command should be executed, is there a more specific check we could use to check the binding?

**Comment by ekohl** [comment]
Given the repetition of the qpid-config I think a define is in order.

File: manifests/qpid.pp (hunk 2 above)
Comment (pmoravec) [inline manifests/qpid.pp]: Dont know how to write it in puppet, but prior this update, one shall see:

    # qpid-config --ssl-certificate=/etc/pki/katello/qpid_client_striped.crt -b "amqps://localhost:5671" exchanges event -r
    Exchange 'event' (topic)
        bind [*.*] => katello_event_queue
    #

while after the change:

    # qpid-config --ssl-certificate=/etc/pki/katello/qpid_client_striped.crt -b "amqps://localhost:5671" exchanges event -r
    Exchange 'event' (topic)
        bind [compliance.created] => katello_event_queue
        bind [entitlement.created] => katello_event_queue
        bind [entitlement.deleted] => katello_event_queue
        bind [pool.created] => katello_event_queue
        bind [pool.deleted] => katello_event_queue
    #

There is not one-line check if one particular binding exists, I am afraid. Just use grep together with above.

**Comment by chris1984** [comment]
Fixed syntax, I think tests are failing since the other pr is not merged yet.

File: manifests/qpid.pp [hunk 3]
```diff
@@ -18,12 +18,14 @@
     require   => Service['qpidd'],
     logoutput => true,
   } ~>
-  exec { 'bind katello entitlements queue to qpid exchange messages that deal with entitlements':
-    command   => "qpid-config --ssl-certificate ${client_cert} --ssl-key ${client_key} -b 'amqps://localhost:5671' bind event ${candlepin_event_queue} '*.*'",
-    onlyif    => "qpid-config --ssl-certificate ${client_cert} --ssl-key ${client_key} -b 'amqps://localhost:5671' queues ${candlepin_event_queue}",
-    path      => '/usr/bin',
-    require   => Service['qpidd'],
-    logoutput => true,
+  qpid::bind_event {'entitlement.created':
```
Comment (ekohl) [inline manifests/qpid.pp]: I think you should pass in the ```candlepin_event_queue``` as a parameter so the ```qpid::bind_event``` stays generic

Comment (chris1984) [inline manifests/qpid.pp]: @ekohl would it look like this example from the puppet docs with candlepin_event_queue being in the spot port is:

apache::vhost {'homepages':
  port    => 8081,
  docroot => '/var/www-testhost',
}

Comment (ekohl) [inline manifests/qpid.pp]: Currently you named the parameter ```candlepin_event_queue``` so it'd look as follows:
```puppet
qpid::bind_event { 'entitlement.created':
  candlepin_event_queue => $candlepin_event_queue,
}
```

But I'd rename the parameter to just ```queue``` so the second line becomes ```queue => $candlepin_event_queue```

**Comment by chris1984** [comment]
@ekohl thanks for the help :) change the formatting and param name. Will change it in the other PR so it reflects as well. Once the other one is good and ready to be merged I think this one should be good then?

**Comment by timogoebel** [comment]
@chris1984 : This change looses the `--ssl-key ${client_key}` parameter when you call `qpid-config` if I read the code right. If I'm not mistaken, `qpid-config` needs both a cert and key to connect correctly.

**Comment by chris1984** [comment]
@timogoebel I was under that impression at first too and was using cert/key but after talking to one of the mrg/qpid guys if we pass it the qpid_client_striped.crt we can skip the key. 

Here is an example of it:

[root@satellite1 ~]# qpid-config --ssl-certificate $CERT -b "amqps://localhost:5671" exchanges
Type      Exchange Name       Attributes
==================================================
direct                        --replicate=none
direct    C.dq                --durable
direct    amq.direct          --durable --replicate=none
fanout    amq.fanout          --durable --replicate=none
headers   amq.match           --durable --replicate=none
topic     amq.topic           --durable --replicate=none
direct    celery              --durable
fanout    celery.pidbox      
topic     celeryev            --durable
topic     event               --durable
direct    qmf.default.direct  --replicate=none
topic     qmf.default.topic   --replicate=none
topic     qpid.management     --replicate=none
direct    resource_manager    --durable
[root@satellite1 ~]#

**Comment by pmoravec** [comment]
Indeed, both options of `--ssl-*` parameters shall work. This one is shorter (1 param instead of 2), used also elsewhere (see e.g. https://github.com/Katello/katello-packaging/blob/master/katello/katello-debug.sh#L81).

File: manifests/qpid.pp [hunk 4]
```diff
@@ -18,12 +18,24 @@
...
   } ~>
-  exec { 'bind katello entitlements queue to qpid exchange messages that deal with entitlements':
-    command   => "qpid-config --ssl-certificate ${client_cert} --ssl-key ${client_key} -b 'amqps://localhost:5671' bind event ${candlepin_event_queue} '*.*'",
-    onlyif    => "qpid-config --ssl-certificate ${client_cert} --ssl-key ${client_key} -b 'amqps://localhost:5671' queues ${candlepin_event_queue}",
-    path      => '/usr/bin',
-    require   => Service['qpidd'],
-    logoutput => true,
+  qpid::bind_event { 'entitlement.created':
+    ssl_cert => '/etc/pki/katello/qpid_client_striped.crt'
+    queue => $candlepin_event_queue,
```
Comment (stbenjam) [inline manifests/qpid.pp]: Please align the hash rocks under one another

File: manifests/qpid.pp [hunk 5]
```diff
@@ -18,12 +18,24 @@
...
     logoutput => true,
   } ~>
-  exec { 'bind katello entitlements queue to qpid exchange messages that deal with entitlements':
-    command   => "qpid-config --ssl-certificate ${client_cert} --ssl-key ${client_key} -b 'amqps://localhost:5671' bind event ${candlepin_event_queue} '*.*'",
-    onlyif    => "qpid-config --ssl-certificate ${client_cert} --ssl-key ${client_key} -b 'amqps://localhost:5671' queues ${candlepin_event_queue}",
-    path      => '/usr/bin',
-    require   => Service['qpidd'],
-    logoutput => true,
+  qpid::bind_event { 'entitlement.created':
+    ssl_cert => '/etc/pki/katello/qpid_client_striped.crt'
```
Comment (stbenjam) [inline manifests/qpid.pp]: This needs a comma after it

**Comment by stbenjam** [comment]
The puppet-qpid change was merged, so you can continue working on this one

**Comment by chris1984** [comment]
@ekohl @stbenjam can I get a review when you get a chance please. Tests are green.

File: manifests/qpid.pp [hunk 6]
```diff
@@ -18,12 +18,24 @@
...
     logoutput => true,
   } ~>
-  exec { 'bind katello entitlements queue to qpid exchange messages that deal with entitlements':
-    command   => "qpid-config --ssl-certificate ${client_cert} --ssl-key ${client_key} -b 'amqps://localhost:5671' bind event ${candlepin_event_queue} '*.*'",
-    onlyif    => "qpid-config --ssl-certificate ${client_cert} --ssl-key ${client_key} -b 'amqps://localhost:5671' queues ${candlepin_event_queue}",
-    path      => '/usr/bin',
-    require   => Service['qpidd'],
-    logoutput => true,
+  qpid::bind_event { 'entitlement.created':
+    ssl_cert => '/etc/pki/katello/qpid_client_striped.crt',
```
Comment (ekohl) [inline manifests/qpid.pp]: You could use ```$client_cert``` here (and below).

File: manifests/qpid.pp [hunk 7]
```diff
@@ -18,12 +18,24 @@
     require   => Service['qpidd'],
     logoutput => true,
   } ~>
-  exec { 'bind katello entitlements queue to qpid exchange messages that deal with entitlements':
-    command   => "qpid-config --ssl-certificate ${client_cert} --ssl-key ${client_key} -b 'amqps://localhost:5671' bind event ${candlepin_event_queue} '*.*'",
-    onlyif    => "qpid-config --ssl-certificate ${client_cert} --ssl-key ${client_key} -b 'amqps://localhost:5671' queues ${candlepin_event_queue}",
-    path      => '/usr/bin',
-    require   => Service['qpidd'],
-    logoutput => true,
+  qpid::bind_event { 'entitlement.created':
```
Comment (ekohl) [inline manifests/qpid.pp]: You could even simplify this:
```puppet
qpid::bind_event { ['entitlement.created', 'entitlement.deleted', 'pool.created', 'pool.deleted', 'compliance.created']:
  ssl_cert => $client_cert,
  queue    => $candlepin_event_queue,
}
```

That would even allow you to parametrize which events to bind to.

Now I am wondering (and we didn't catch this in the other PR): where did the ```--ssl-key ${client_key}``` parameter go?

Comment (ekohl) [inline manifests/qpid.pp]: Disregard the last bit, I see there was already a discussion about it.

Comment (chris1984) [inline manifests/qpid.pp]: @ekohl After talking to @pmoravec with passing the striped client crt like here, https://github.com/Katello/katello-packaging/blob/master/katello/katello-debug.sh#L81 we dont need the key.

**Comment by chris1984** [comment]
@ekohl Will work on an updated commit with the suggested updates

**Comment by ekohl** [comment]
Another question: will this work for any katello version or just nightly?

**Comment by chris1984** [comment]
@stbenjam @ekohl new changes submitted, I tested the manual commands on downstream 6.2 which is Katello 3.0 and when working on the PR i was testing it on Katello 3.3 since nightly was broken for a while.

**Pull request merged**